### PR TITLE
fix: Issue 26751 QR barcode scanner

### LINKED ISSUE #26751: [Bug]: Send barcode scanner does not work
### Describe the bug

When trying to use the barcode scanner in the send flow on my laptop (macbook pro), I get an error saying the recipient address is invalid.

https://github.com/user-attachments/assets/08b5a2e6-aad7-44f6-8b23-39b479e4afd1

### Expected behavior

_No response_

### Screenshots/Recordings

_No response_

### Steps to reproduce

1. Click send from the wallet screen
2. Click the scan barcode icon in the to address field
3. Give camera access to the extension/browser
4. Show barcode in the camera

### Error messages or log output

_No response_

### Detection stage

In production (default)

### Version

12.0.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by ejwessel**
## **Description**
QR scanning broke some time ago. This PR fixes scanning QR codes when provided a QR code from either mobile or from extension by extracting just the address.

Instead of using various splits and conditionals, I use regex capture groups to extract the address.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27002?quickstart=1)

## **Related issues**

Fixes:
https://github.com/MetaMask/metamask-extension/issues/26751
https://github.com/MetaMask/metamask-extension/issues/21766

## **Manual testing steps**

1. select 'send'
2. select the scan button in the destination address `to` field
3. either select a QR code provided by the mobile app or provided by extension that contains a valid ethereum address
4. validate that the address populates and no error is shown.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/modals/qr-scanner/qr-scanner.component.js [hunk 1]
```diff
@@ -30,8 +30,8 @@ const parseContent = (content) => {
   // Ethereum address links - fox ex. ethereum:0x.....1111
   if (content.split('ethereum:').length > 1) {
     type = 'address';
-    values = { address: content.split('ethereum:')[1] };
-
+    // uses regex capture groups to match and extract address while ignoring everythign else
```
Comment (darkwing) [inline ui/components/app/modals/qr-scanner/qr-scanner.component.js:33]: Nit: `everythign` -> `everything`

File: ui/components/app/modals/qr-scanner/qr-scanner.component.js [hunk 2]
```diff
@@ -30,8 +30,8 @@ const parseContent = (content) => {
...
+    values = { address: content.match(/.*:(0x[0-9a-fA-F]{40})(?:@.*)?/)[1] }
```
Comment (darkwing) [inline ui/components/app/modals/qr-scanner/qr-scanner.component.js:34]: I hate pointing this out but should we have this regex in a function so we can add a test?  Also, is there a product consideration here?  i.e. if the string ends in `0x89`, and the user is on Mainnet, should something happen?

Comment (ejwessel) [reply ui/components/app/modals/qr-scanner/qr-scanner.component.js:34]: +1 on test

> . if the string ends in 0x89, and the user is on Mainnet, should something happen?

perhaps though I feel that's outside the scope of the bug.

**Review by jclancy93 — DISMISSED** [review]

**Review by darkwing — DISMISSED** [review]
Works as expected!  Thank you for adding tests!

File: yarn.lock [hunk 3]
```diff
@@ -14403,9 +14403,9 @@ __metadata:
   linkType: hard
 
 "caniuse-lite@npm:^1.0.30001109, caniuse-lite@npm:^1.0.30001587, caniuse-lite@npm:^1.0.30001599":
-  version: 1.0.30001600
-  resolution: "caniuse-lite@npm:1.0.30001600"
-  checksum: 10/4c52f83ed71bc5f6e443bd17923460f1c77915adc2c2aa79ddaedceccc690b5917054b0c41b79e9138cbbd9abcdc0db9e224e79e3e734e581dfec06505f3a2b4
+  version: 1.0.30001660
+  resolution: "caniuse-lite@npm:1.0.30001660"
+  checksum: 10/5d83f0b7e2075b7e31f114f739155dc6c21b0afe8cb61180f625a4903b0ccd3d7591a5f81c930f14efddfa57040203ba0890850b8a3738f6c7f17c7dd83b9de8
```
Comment (DDDDDanica) [inline yarn.lock:14408]: Question: is this expected since there's no package change?

**Review by darkwing — DISMISSED** [review]

**Review by micaelae — APPROVED** [review]

**Review by jclancy93 — APPROVED** [review]

**Review by DDDDDanica — APPROVED** [review]

**Pull request merged**